### PR TITLE
SB-1175 Fix UserController endpoints

### PR DESCRIPTION
**Pull request by steve-todorov**

### CODE REVIEW

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/users/UserController.java [hunk 1]
```diff
@@ -52,71 +52,91 @@
...
-    /**
-     * This method exists for testing purposes.
-     */
-    @ApiOperation(value = "Used to retrieve an request param", position = 1)
-    @ApiResponses(value = { @ApiResponse(code = 200, message = "") })
-    @PreAuthorize("authenticated")
-    @GetMapping(value = "/{anyString}",
-            produces = { MediaType.TEXT_PLAIN_VALUE,
-                         MediaType.APPLICATION_JSON_VALUE })
+    @ApiOperation(value = "Used to retrieve an user")
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/users/UserController.java]: Cosmetic: ...`Used to retrieve a user`, not `an`.

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/users/UserController.java [hunk 2]
```diff
@@ -52,71 +52,91 @@
...
+            return getNotFoundResponseEntity(NOT_FOUND_USER, accept);
+        }
 
-        return getSuccessfulResponseEntity("hello, " + param, accept);
+        UserOutput userOutput = UserOutput.fromUser(user);
+        Object body = getUserOutputEntityBody(userOutput, accept);
+        return ResponseEntity.ok(body);
     }
 
     @ApiOperation(value = "Used to create new user")
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/users/UserController.java]: Cosmetic: `Used to create a new user`.

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/users/UserController.java [hunk 3]
```diff
@@ -133,109 +153,84 @@ public ResponseEntity create(@RequestBody @Validated(UserForm.NewUser.class) Use
...
-    public ResponseEntity update(@RequestBody @Validated(UserForm.ExistingUser.class) UserForm userToUpdate,
+    public ResponseEntity update(@ApiParam(value = "The name of the user", required = true)
+                                 @PathVariable String username,
+                                 @RequestBody @Validated(UserForm.ExistingUser.class) UserForm userToUpdate,
                                  BindingResult bindingResult,
                                  Authentication authentication,
                                  @RequestHeader(HttpHeaders.ACCEPT) String accept)
     {
+        // If the {username} is admin - don't allow editing!
+        if (StringUtils.equals(username, "admin"))
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/users/UserController.java]: There is no guarantee that there will be exactly one admin user. In fact, if the XML config is not the configured authentication provider (let's say we're using LDAP), the configuration may be such that this user is no longer present, as the owner of the service may have decided to delete it and have the admins "imported" from LDAP.

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/users/UserController.java [hunk 4]
```diff
@@ -133,109 +153,84 @@ public ResponseEntity create(@RequestBody @Validated(UserForm.NewUser.class) Use
...
+                   produces = { MediaType.TEXT_PLAIN_VALUE,
+                                MediaType.APPLICATION_JSON_VALUE })
     @ResponseBody
-    public ResponseEntity delete(@ApiParam(value = "The name of the user") @PathVariable String name,
+    public ResponseEntity delete(@ApiParam(value = "The name of the user") @PathVariable String username,
                                  Authentication authentication,
                                  @RequestHeader(HttpHeaders.ACCEPT) String accept)
     {
-        if (!(authentication.getPrincipal() instanceof SpringSecurityUser))
+        if (StringUtils.equals(username, "admin"))
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/users/UserController.java]: Same as above.

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/users/UserController.java (hunk 3 above)
Comment (fuss86) [reply strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/users/UserController.java]: Please review SB-868, @carlspring

**Comment by carlspring** [comment]
Thanks!

**Pull request merged**